src/main/java/services/RestoreFactory.java

```diff
@@ -337,22 +329,38 @@ private InputStream getRestoreXmlHelper(String restoreUrl, HqAuth auth) {
         return stream;
     }
 
-    public static String getRestoreUrl(String host, String domain, boolean overwriteCache){
-        String url = host + "/a/" + domain + "/phone/restore/?version=2.0";
-        if (overwriteCache) {
-            url += "&overwrite_cache=true";
+    private String getSyncToken(String username) {
+
+        if (username == null) {
+            return null;
+        }
+        SqliteIndexedStorageUtility<User> storage = getSqlSandbox().getUserStorage();
+        Vector<Integer> users = storage.getIDsForValue(User.META_USERNAME, username);
+        //should be exactly one user
+        if (users.size() != 1) {
+            return null;
         }
-        return url;
-    }
 
-    public String getRestoreUrl(String host, String domain, String username, boolean overwriteCache) {
-        String url = host + "/a/" + domain + "/phone/restore/?as=" + username + "@" +
-                domain + ".commcarehq.org&version=2.0";
+        return storage.getMetaDataFieldForRecord(users.firstElement(), User.META_SYNC_TOKEN);
+    }
 
+    public String getRestoreUrl(boolean overwriteCache) {
+        StringBuilder builder = new StringBuilder();
+        builder.append(host);
+        builder.append("/a/");
+        builder.append(domain);
+        builder.append("/phone/restore/?version=2.0");
         if (overwriteCache) {
-            url += "&overwrite_cache=true";
+            builder.append("&overwrite_cache=true");
```

Comment:
this overwrites all of HQs caching of restores. Is there a reason you do this, and not let HQ handle the way it deals with its own caching? I was under the impression that this param was only called for debugging purposes. 
If you hit the restore URL without a `since` param, I think it should return what you expect (i.e. it will generate a new restore payload, or fetch the restore from the cache if it can)